src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -486,7 +467,11 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
           "I loaded all of {} checkpointed ids from the DB and will proceed to fetch updated accounts information from the chain",
           checkpoints.size
         )
-        AccountsProcessor.process(checkpoints, onlyProcessLatest = true).map(_ => Done)
+        // here we need to get missing bakers for the given block
+        tezosNodeOperator.getBakersByHash(checkpoints.values.map(_._1).toList).flatMap { bakers =>
```

Comment:
We can try avoiding to reload data from the node, by using the stored database records.
All the info should be there already, it would imply
* passing a different data type to `AccountsProcessor.process`
* convert the node operator call to a database operator one
* remove the custom baker-rolls fetcher that loads by `block-hash`

In any case, should we not follow this path, I'd suggest adapting the existing node-fetcher to only work on block-hash, instead of writing an -  essentially - duplicate one. I can imagine adapting the client's call to extract the block's hash before fetching the node data for bakers' rolls

Reply:
made the changes, removed my fetcher - it was only used during reading from checkpoints, but also had to move up fetching voting data before the checkpoints to ensure that if the writing checpoints will be stopped in the middle we would have corresponding bakers data in the tables

Reply:
But am I wrong or now we could end up with stored votes and then a failing save on checkpoints or blocks, leaving "ophaned" records of votes?

Won't those vote be rewritten again on lorre's restart? Please make sure all is kept consistent, let me know if I can help to think of a solution.

Reply:
I think you are right, I'm going to change that, because the previous solution seems more reasonable, although not perfect

Reply:
Yes, I agree, it might've not being apparent, but votes might get lost on failure, at times... 
But making this step more robust might not be that easy